tests/ui/features/bootstrap/FilesContext.php

```
@@ -158,15 +158,25 @@ public function theFilesPageIsReloaded() {
 	}
 
 	/**
-	 * @When /^I create a folder with the name ((?:'[^']*')|(?:"[^"]*"))$/
+	 * @When /^I create a folder with the (invalid|)\s?name ((?:'[^']*')|(?:"[^"]*"))$/
 	 *
+	 * @param string $invalid contains "invalid"
+	 * 						  if the folder creation is expected to fail
 	 * @param string $name enclosed in single or double quotes
 	 * @return void
 	 */
-	public function iCreateAFolder($name) {
+	public function iCreateAFolder($invalid, $name) {
 		// The capturing group of the regex always includes the quotes at each
 		// end of the captured string, so trim them.
-		$this->createAFolder(trim($name, $name[0]));
+		$name = trim($name, $name[0]);
+		try {
+			$this->createAFolder($name);
```

If expected to be invalid but for some reason there is no exception thrown by ``createAFolder`` then should we throw an exception here?
So that we know straight away that the step did not have the expected problem.

---

change bayo